src/realm-settings/KeysProvidersTab.tsx

```diff
@@ -130,9 +133,23 @@ export const KeysTabInner = ({ components, refresh }: KeysTabInnerProps) => {
     setLiveText(t("onDragCancel"));
   };
 
-  const onDragFinish = (itemOrder: string[]) => {
-    setItemOrder(["data", ...itemOrder.filter((i) => i !== "data")]);
+  const onDragFinish = async (itemOrder: string[]) => {
+    setItemOrder([...itemOrder]);
```

Comment:
Same here, I don't think this needs to be spread.
```suggestion
    setItemOrder(itemOrder);
```